bee-protocol/src/workers/mod.rs

```diff
@@ -43,13 +43,16 @@ pub(crate) use responder::{
 pub(crate) use solidifier::{MilestoneSolidifierWorker, MilestoneSolidifierWorkerEvent};
 pub(crate) use status::StatusWorker;
 
-use bee_network::NetworkEventReceiver;
+use bee_autopeering::event::EventRx as AutopeeringEventRx;
+use bee_gossip::NetworkEventReceiver as NetworkEventRx;
 use bee_runtime::node::{Node, NodeBuilder};
+use peer::PeerManagerConfig;
 
 pub fn init<N: Node>(
     config: config::ProtocolConfig,
-    network_id: u64,
-    events: NetworkEventReceiver,
+    network_id: (String, u64),
```

Comment:
Why is this a different type than `NodeConfig.identify`? Would it make sense for this to be a struct with named members that is shared throughout?

Reply:
yes, I guess so. This struct would need to be in `bee-common` I guess so we don't run into cyclic dep issues, but I that would have to be done through another PR. 

Please review again...I refactored this again into a named struct called `NetworkSpec`.